osu.Game/Beatmaps/FlatFileWorkingBeatmap.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using osu.Framework.Audio.Track;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Beatmaps.Formats;
+using osu.Game.IO;
+using osu.Game.Rulesets;
+using osu.Game.Skinning;
+
+namespace osu.Game.Beatmaps
+{
+    /// <summary>
+    /// A <see cref="WorkingBeatmap"/> which can be constructed directly from a .osu file, providing an implementation for
+    /// <see cref="WorkingBeatmap.GetPlayableBeatmap(osu.Game.Rulesets.IRulesetInfo,System.Collections.Generic.IReadOnlyList{osu.Game.Rulesets.Mods.Mod})"/>.
+    /// </summary>
+    public class FlatFileWorkingBeatmap : WorkingBeatmap
```

Comment:
This is a helper class, used both in this new code but also has matching implementations in osu-tools and osu-difficulty-calculator. I hope to eventually change the `rulesetProvider` ctor arg to an `IRulesetStore` or something similar as we've discussed having last week.